src/corelibs/U2Algorithm/src/misc/EnzymeModel.h

```diff
@@ -76,6 +76,10 @@ class U2ALGORITHM_EXPORT EnzymeSettings {
 
 class U2ALGORITHM_EXPORT EnzymeData : public QSharedData {
 public:
+    bool operator==(const EnzymeData& other) const {
+        return id == other.id && seq == other.seq;
```

Comment:
There should be no 2 enzymes with the same id registered in the system at the same moment

Reply:
It could be and it legally happened - see "BarI". This enzyme has two possible cut pairs and we show them as separate enzyme.

Reply:
Lets do not inherit other system errors: ID is a unique identifier in UGENE - it's along should be enough for the equality comparison.

I would suggest just drop the duplicate: AFAIU BarI duplicate is a plain rev-compl of the original one.

Are there any other enzymes like this? If yes and they are different we should reconsider our enzyme model: keep ids unique and introduce a new dimension for the variation like this.

Reply:
Even if this is the only enzyme duplicated, we need to handle it, can't just drop it. I can split the only `id` field to `id` and `name`. Should I do it now? We've been living with it for a decade, maybe, at lease in separate issue?

Reply:
The problem I mentioned is in the new public method you added: if you do not introduce the wrong function I do not ask to fix the old issue as a part of this PR: we can make it a separate PR.

Reply:
OK